owasp-top10-2017-apps/a5/tictactoe/src/app.js

```diff
@@ -153,41 +154,48 @@ app.post('/login', async (req, res) => {
     const username = req.body.username
     const password = req.body.password
     const salt = await db.getPasswordSalt(username)
-    if (salt === undefined){
+    const type = await db.getType(username)
+    if (type === undefined) {
+        res
+            .status(400)
```

Comment:
Since `type` was not in the request sent by the client, we should not use a 400 error. A 500 seems more suitable in this case (because type is only undefined if our database is corrupted, right?)